### PR TITLE
tests: untrimmed stdout assertion (fix #1235)

### LINKED ISSUE #1235: Testing framework hides issues
The testing framework strips trailing spaces/newlines from the output, causing things such as `test_stdin_show_tabs()` for `cat` to report success despite printing an extra newline.

### PR DESCRIPTION
**Pull request by rinconjc**
Also fixed issues in cat and join uncovered after removing the
trimming above.

### CODE REVIEW

File: tests/test_realpath.rs [hunk 1]
```diff
@@ -4,20 +4,20 @@ use common::util::*;
...
-    ucmd.arg(dir).run().stdout_is(at.root_dir_resolved());
+    ucmd.arg(dir).run().stdout_is(at.root_dir_resolved()+"\n");
 }
 
 #[test]
 fn test_long_redirection_to_root() {
     // Create a 255-character path to root
     let dir = path_concat!("..", ..85);
-    new_ucmd!().arg(dir).run().stdout_is(get_root_path());
+    new_ucmd!().arg(dir).run().stdout_is(get_root_path().to_owned() +"\n");
```
Comment (Arcterus) [inline tests/test_realpath.rs]: For all of these, can you make sure the `+` is separated on both sides by a space?

**Comment by Arcterus** [comment]
IIRC I noticed this a while ago but got caught up with something and forgot to fix the issue.  Good catch!

File: tests/test_realpath.rs (hunk 1 above)
Comment (rinconjc) [reply tests/test_realpath.rs]: no problem, it's done.
```suggestion
    new_ucmd!().arg(dir).run().stdout_is(get_root_path().to_owned() +"\n");
```

**Comment by Arcterus** [comment]
Thanks!

**Pull request merged**